Simulator/Layouts/Neuro/FixedLayout.cpp

```diff
@@ -12,14 +12,39 @@
 #include "Util.h"
 
 // TODO: I don't think that either of the constructor or destructor is needed here
-FixedLayout::FixedLayout() : Layout()
+FixedLayout::FixedLayout() : gridLayout_(true), Layout()
 {
 }
 
 FixedLayout::~FixedLayout()
 {
 }
 
+void FixedLayout::setup()
+{
+   // Base class allocates memory for: xLoc_, yLoc, dist2_, and dist_
+   // so we call its method first
+   Layout::setup();
+
+   // Initialize neuron locations memory, grab global info
+   initVerticesLocs();
+
+   // computing distance between each pair of vertices given each vertex's xy location
+   for (int n = 0; n < numVertices_ - 1; n++) {
+      for (int n2 = n + 1; n2 < numVertices_; n2++) {
+         // distance^2 between two points in point-slope form
+         (*dist2_)(n, n2) = ((*xloc_)[n] - (*xloc_)[n2]) * ((*xloc_)[n] - (*xloc_)[n2])
+                            + ((*yloc_)[n] - (*yloc_)[n2]) * ((*yloc_)[n] - (*yloc_)[n2]);
+
+         // both points are equidistant from each other
+         (*dist2_)(n2, n) = (*dist2_)(n, n2);
+      }
+   }
+
+   // take the square root to get actual distance (Pythagoras was right!)
+   (*dist_) = sqrt((*dist2_));
```

Comment:
So, why does the superclass own the `dist_` and `dist2_` variables, but they're initialized in the subclass?

Reply:
This is a good question. Both `dist_` and `dist2_` are calculated from `xloc_` and `yloc_`, and since their initialization is different in the 911 system, I had to move the initialization of all those arrays to the subclasses. I realize now that this probably breaks `DynamicLayout`, which I am not sure what it does exactly but looking at its code it could inherit from `FixedLayout` and just override `printParameters()`, `generateVertexTypeMap()`, and `initStarterMap()`.

And the question is how to handle class members that are common to all models but whose initialization differs. Should they be part of the base class, or moved to their respective derived classes?